src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,217 @@
+ * It will be placed according to the `position` property.
+ *
+ * ## Accessibility
+ *
+ * The component places necessary aria attribute on the `button` element:
+ *
+ * * `aria-expanded`: set to `true` when popover is open, `false` when it is closed.
+ *
+ * When the popover is open, pressing `esc` will close the popover and focus the `button` element.
+ *
+ * ## Usage
+ *
+ * ```html
+ * <cc-popover>
+ *   <span slot="button-text">Click me</span>
+ *   <div>This is a content to be displayed when button is clicked.</div>
+  * </cc-popover>
+ * ```
+ *
+ * @cssdisplay block
+ *
+ * @event {CustomEvent} cc-popover:open - Fires whenever the popover is opened.
+ * @event {CustomEvent} cc-popover:close - Fires whenever the popover is closed.
+ *
+ * @slot - The area containing the content of the popover.
+ * @slot button-text - The area containing the button text.
+ *
+ * @cssprop {Number} --cc-popover-z-index - Sets the z-index of the floating content (defaults: `999`).
+ */
+export class CcPopoverComponent extends LitElement {
+  static get properties () {
+    return {
+      accessibleName: { type: String, attribute: 'accessible-name' },
+      hideText: { type: Boolean, attribute: 'hide-text' },
+      icon: { type: Object },
+      isOpen: { type: Boolean, attribute: 'is-open', reflect: true },
+      position: { type: String },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {string|null} Sets the accessibleName property of the underlying `cc-button` element. CAUTION: The accessible name should always start with the visible text if there is one. For instance "add to estimation - NodeJS XS" */
+    this.accessibleName = null;
+
+    /** @type {boolean} Whether the button text should be hidden. */
+    this.hideText = false;
+
+    /** @type {IconModel|null} Sets the button icon. */
+    this.icon = null;
+
+    /** @type {boolean} Whether the popover is opened */
+    this.isOpen = false;
+
+    /** @type {PopoverPosition} Sets the position of the popover relative to the `button` element. */
+    this.position = 'bottom-left';
+
+    /** @type {Ref<HTMLElement>} */
+    this._contentRef = createRef();
+
+    /** @type {Ref<HTMLElement>} */
+    this._buttonRef = createRef();
+
+    this._onOutsideClickHandler = new EventHandler(window, 'click', (event) => {
+      const contentElement = this._contentRef.value;
+      if (contentElement != null && !event.composedPath().includes(contentElement)) {
+        this.close();
+      }
+    });
+    this._onEscapeKeyHandler = new EventHandler(this, 'keydown', (event) => {
+      if (event.key === 'Escape') {
+        this.close();
+      }
+    });
+  }
+
+  /**
+   * Opens the popover.
+   */
+  open () {
+    if (!this.isOpen) {
+      this.isOpen = true;
+      dispatchCustomEvent(this, 'open');
+    }
+  }
+
+  /**
+   * Closes the popover.
+   */
+  close () {
+    if (this.isOpen) {
+      this.isOpen = false;
+      this._buttonRef.value?.focus();
```

Comment:
**suggestion:** considering this is a public method that could be used from outside in case we don't foresee, I wonder if we could add a `shouldMoveFocus` boolean flag (`true` by default) to this methods and only move focus when it's true.

For instance, imagine we programatically close several popovers at once, where should the focus go?
Same idea, imagine we have a button that is used to close all popovers at once, the focus should actually remain on that button.

These cases are not very likely/common but I think it's an easy to do so we should go for it.

Reply:
Very good suggestion. Thank you